students/CCSt130/lesson03/mail_room_pt1.py

```diff
@@ -0,0 +1,500 @@
+                all_donations = add_new_donor(all_donations, a_new_donor)
+                # print()
+
+            elif usr_input == 'Q' or usr_input == 'q':
+                break
+
+            else: 
+                print()
+                print("Invalid entry--try again!")
+                print()
+
+        print()
+        print("### Program Exit ###")
+
+def new_donor_input():
+    """ Takes user's choice of donor and returns it. """
+    while True:
+        # Prompt user for input
+        input_msg = "Please type the name of an Organization: "
+        # Capture user choice 
+        usr_input = str(input(input_msg))
+        print()
+        # Could put a loop here to allow correction
+        print("You entered: {}".format(usr_input))
+
+        # Strip white space for comparison
+        input_strip = usr_input.replace(' ', '')
+
+        # Filter out non-alpha entry            
+        if input_strip.isalpha():
+            break 
+        else:
+            print()                
+            print("Invalid entry--try again!")
+            print()
+
+    return(usr_input)
+
+def donor_sub_menu(upd_donation_list): # updated donation list
+        """ Displays 'nested' menu of donors. """
+        while True:
+            # Displaying menu header
+            print()
+            print()
+            print(48 * '*')
+            print()
+            print(17 * '*', "Choose Donor", 17 * '*')
+
+            list_donor_names(upd_donation_list)
+            
+            # print()
+            # print("Enter 'Q' to Quit Sub-Menu")        
+            print()
+            print(48 * '*')
+
+            # Launch function to capture user choice 
+            new_donor = new_donor_input()
+
+            print()
+            return(new_donor)
+
+def prn_all_donations(donations):
+    """ Prints formatted donation details for all donors. """
+
+    # Sort based on first element
+    donations.sort()
+    # Find length of list
+    donor_len = len(donations)
+    # Column labels
+    header1 = ['Date', 'Donor', 'Amount']
+
+    print()
+    print("Donation History:") 
+    print()
+    # Print header
+    print('{:<13} {:<36} {}'.format(*header1))
+    print(65 * '-')
+   
+    d = 0 # While loop counter and index
+    
+    # Print formatted list of each donation date, donor, amount of donation
+    while(d < donor_len):    
+        # Seems best to pass formatted date to a new name
+        # pull this out into a function
+        formatted_date = (datetime.date(donations[d][0], donations[d][1], donations[d][2]))
+        # For an array, especially a 2D array, the .format syntax seems easier to implement        
+        # Print date first with 4 spaces on the end
+        print('{}'.format(formatted_date), end = "    ")
+        # Print donor name and amount
+        print('{:<36} ${:>13,.2f}'.format(donations[d][3], donations[d][4]))
+    
+        d+=1 # Increment counter
+
+    print()        
```

Comment:
option # 1 should be printing a report that has following columns and look similar to this:
```
DONOR               | TOTAL GIVEN       | NUMBER OF GIFTS   | AVERAGE GIFTS
-----------------------------------------------------------------------------
James Smith               2527.75                3               842.58
```
it should also be sorted by total given